src/main.rs

```diff
@@ -323,7 +343,11 @@ struct Args {
     pub tls_ca_paths: Option<Vec<PathBuf>>,
 
     /// If provided, overrides the default heartbeat interval of every 30 seconds. Provided value is interpreted as seconds.
-    #[arg(long = "heartbeat-interval-seconds", env = "WASMCLOUD_HEARTBEAT_INTERVAL", value_parser = parse_duration_secs, hide = true)]
+    #[arg(long = "heartbeat-interval-seconds", env = "WASMCLOUD_HEARTBEAT_INTERVAL_SECONDS", value_parser = parse_duration_secs, hide = true, conflicts_with = "heartbeat_interval")]
```

Comment:
This change is _slightly_ problematic in terms of backward compatibility, i.e. if someone was already using the `WASMCLOUD_HEARTBEAT_INTERVAL` environment variable with a value like `10`, we would parse that as 10 **milliseconds** instead of **seconds**.

I could leave the env var as is, and call the new env var something like `WASMCLOUD_HEARTBEAT_INTERVAL_TIME` 😅

Reply:
Yeah interesting -- I kind of prefer getting this right/reasonable as early as possible (if we're going to break, break sooner rather than later and guide people forward)...  BUT if someone had this specified, going from 10 seconds to 10ms is quite intense.

Maybe it make sense to write a parser that *defaults* to seconds if unspecified but uses humantime otherwise?

Reply:
That makes sense. I could extend `parse_duration` to take an optional `unit` parameter that defaults to `ms` but in this case I can pass in `sec` as an argument?

Reply:
Yeah that would be great -- I'm not sure how you could make that work with how you call it via `clap` but that sounds reasonable to me

Reply:
Yeah that would be great -- I'm not sure how you could make that work with how you call it via `clap` but that sounds reasonable to me

Reply:
Ah okay, can't do partial application or something for functions in the clap macro. I'm just going to update the `parse_duration_secs` method to _either_ parse `humantime` _or_ keep existing behavior i.e. get duration from `u64` assuming seconds.

Reply:
This appears to work, but IMO makes things less readable
```rust
enum TimeUnit {
    Ms,
    Sec,
}

fn parse_duration(arg: &str, unit: Option<TimeUnit>) -> anyhow::Result<Duration> {
    if let Ok(duration) = humantime::Duration::from_str(arg) {
        return Ok(duration.into());
    }
    match unit {
        Some(TimeUnit::Sec) => {
            if let Ok(secs) = arg.parse::<u64>() {
                return Ok(std::time::Duration::from_secs(secs));
            }
        }
        _ => {
            if let Ok(millis) = arg.parse::<u64>() {
                return Ok(std::time::Duration::from_millis(millis));
            }
        }
    }
    Err(anyhow::anyhow!("Invalid duration: '{}'. Expected a duration like '5s', '1m', '100ms', or milliseconds as an integer.", arg))
}
```
and in `clap`:
```rust
    /// If provided, overrides the default heartbeat interval of every 30 seconds. Provided value is interpreted as seconds.
    #[arg(long = "heartbeat-interval", env = "WASMCLOUD_HEARTBEAT_INTERVAL", value_parser = |s: &str| parse_duration(s, Some(TimeUnit::Sec)), hide = true)]
    heartbeat_interval: Option<Duration>,
```
but we need to add that closure everywhere `|s: &str| parse_duration(s, Some(TimeUnit::Ms))` which doesn't feel worth it to me

Reply:
I think you're really clsoe to something usable there! That said, it might be simpler to just write *really light* wrappers that use that big fn -- `parse_time_fallback_seconds` might be a decent name?

Reply:
I like that idea, thanks! Pushed https://github.com/wasmCloud/wasmCloud/pull/3440/commits/b138fe2c9bfa02b1e69e8281337f14f7d6945552